### ISSUE TITLE
root.AgentAct freezes Unity

### ISSUE THREAD

**Issue by aliosmanbusiness** [Unity-Technologies/ml-agents#4136]
While I try to train my agent on tensorflow, Unity freezes during training. Here is profiler:

![Screen Shot 2020-06-18 at 12 52 00](https://user-images.githubusercontent.com/54140117/85006545-ec719700-b162-11ea-87e9-67f3750220dd.png)

Here is my configuration for agent:

behaviors:
  MuniAgent:
    trainer_type: ppo
    hyperparameters:
      batch_size: 10
      buffer_size: 100
      learning_rate: 3.0e-4
      beta: 5.0e-4
      epsilon: 0.2
      lambd: 0.99
      num_epoch: 3
      learning_rate_schedule: linear
    network_settings:
      normalize: false
      hidden_units: 128
      num_layers: 2
    reward_signals:
      extrinsic:
        gamma: 0.99
        strength: 1.0
    max_steps: 500000
    time_horizon: 64
    summary_freq: 10000

**Comment by MrForExample** [comment]
You need put more information to let others help you with this issue, follow the template of the post new issue, show version of your package, at least show others how to reproduce the issue.

**Comment by chriselion** [comment]
Does this happen with the provided example scenes? If not, the problem is likely due to a bug in your code. If you post it, I can try to help debug it

**Comment by aliosmanbusiness** [comment]
@chriselion Thank you for your help, I found that my code was responsible for the freezing. I used recursive function for calculation it was working while playing normally, yet it produces freezing while training.
